src/java/io/bazel/rulesscala/test_discovery/DiscoveredTestSuite.scala

```diff
@@ -108,11 +114,11 @@ object PrefixSuffixTestDiscoveringSuite {
     .map(_.getName)
     .toList
 
-  private def archiveInputStream() =
+  private def archiveInputStream(archivePath: String) =
     new JarInputStream(new FileInputStream(archivePath))
 
-  private def archivePath: String =
-    System.getProperty("bazel.discover.classes.archive.file.path")
```

Comment:
is this going to break anyone, or is this only expected to be set by a rule? If a rule, is it our rule? should we use `io.bazel.rules_scala` or something to distinguish? Can we add a comment in any case to ease reading the code?

Reply:
Won't break anyone
 our JUnit rule
I'm actually thinking of factoring out the JUnit "discover test classes" mechanism (which this is a part of) since it's not scala related but JUnit related.
When we added this we couldn't do this refactoring due to how scala and java interoped. Due to this I'd rather not change the name.
Re the comment- If one will search the repo for this text he will find it in the bzl so I think it's ok for now but if you feel strongly about it I'll add the comment

Reply:
done